router/internal/persistedoperation/operationstorage/cache.go

```diff
@@ -14,6 +16,25 @@ type OperationsCache struct {
 	Cache *ristretto.Cache[string, []byte]
 }
 
+func NewOperationsCache(cacheSize int64) (*OperationsCache, error) {
+	if cacheSize <= 0 {
+		return &OperationsCache{}, nil
+	}
+
+	cache, err := ristretto.NewCache(&ristretto.Config[string, []byte]{
+		// assume an average of persistentAverageCacheEntrySize per operation, then
+		// multiply by 10 to obtain the recommended number of counters
+		NumCounters: (cacheSize * 10) / persistentAverageCacheEntrySize,
```

Comment:
This calculation is very imprecise. We can't make any assumption on average size. I know this existed before and we should come up with something better.

Reply:
Do you have a proposal for how we should create this?

Reply:
You can have a custom cost function in ristretto and count the document bytes.

Reply:
that cost function sounds pretty close to what I'd envision the default cost function to be, and it doesn't address what default num counters we should provide

Reply:
Updated